src/guiFormSpecMenu.cpp

```diff
@@ -343,17 +349,34 @@ void GUIFormSpecMenu::parseList(parserData* data,std::string element)
 		geom.Y = stoi(v_geom[1]);
 
 		s32 start_i = 0;
-		if(startindex != "")
+		if(has_startindex)
 			start_i = stoi(startindex);
 
-		if (geom.X < 0 || geom.Y < 0 || start_i < 0) {
+		v2s32 _imgsize = imgsize;
+		if (has_slot_size_factor && !slot_size_factor.empty()) {
+			f32 size_factor = stof(slot_size_factor);
+			_imgsize.X = (s32)(imgsize.X * size_factor);
+			_imgsize.Y = (s32)(imgsize.Y * size_factor);
+		}
+
+		v2s32 _spacing = spacing;
+		if (has_spacing_factors && v_spacing_factors.size() == 2) {
+			_spacing.X = _imgsize.X + 2 * border
+			+ (s32)(stof(v_spacing_factors[0]) * (spacing.X - imgsize.X - 2 * border));
+			_spacing.Y = _imgsize.Y + 2 * border
+			+ (s32)(stof(v_spacing_factors[1]) * (spacing.Y - imgsize.Y - 2 * border));
```

Comment:
One more indent for each continued new line. (may even require a 3rd line at the end)